### PR TITLE
Restructure both three things

### LINKED ISSUE #4486: Ben Kaduk's TLS Comment 10
@kaduk said:

> Section 4.9
> 
>    > An endpoint cannot discard keys for a given encryption level unless
>    > it has both received and acknowledged all CRYPTO frames for that
>    > encryption level and when all CRYPTO frames for that encryption level
>    > have been acknowledged by its peer.  However, this does not guarantee
> 
> (nit/editorial) I believe that the "both" is meant to apply to "received
> and acknowledged" and "when all CRYTO frames [...] have been
> acknowledged", but the current formulation does not have parallel
> structure between those clauses, so it ends up reading more naturally as
> saying that the "both" refers to "received" and "acknowledged".  I don't
> have a simple suggestion for fixing it, though (it might require a
> significant restructuring to fix), so I'm leaving it here in my comments
> rather than incorporating a fix into my editorial PR.  (Any drastic
> rewording might consider the rest of the paragraph as well, as it seems
> to leave the reader without a clear sense for when it definitely is safe
> to discard old keys.)
> 
>    > Though an endpoint might retain older keys, new data MUST be sent at
>    > the highest currently-available encryption level.  Only ACK frames
>    > and retransmissions of data in CRYPTO frames are sent at a previous
>    > encryption level.  These packets MAY also include PADDING frames.
> 
> Is there anything useful to say about whether it is advisable to make
> retransmission packets have the same size as the original transmission?
> ("No" is a fine answer.)﻿

### PR DESCRIPTION
**Pull request by MikeBishop**
Fixes #4486.  The three points are wordy enough it might be worth making this a bulleted list, however.  Thoughts?

### CODE REVIEW

**Review by martinthomson — APPROVED** [review]
It's really tempting to note that you can just lose packets containing CRYPTO frames, which will never get acknowledged.  More generally, loss or spurious retransmissions would result in this text being technically incorrect.    Should this talk about the data contained in CRYPTO frames instead?

Other than that, I really like the phrasing.

File: draft-ietf-quic-tls.md [hunk 1]
```diff
@@ -836,12 +836,13 @@ encryption level as the packet being acknowledged.  Thus, it is possible that
...
-An endpoint cannot discard keys for a given encryption level unless it has both
-received and acknowledged all CRYPTO frames for that encryption level and when
-all CRYPTO frames for that encryption level have been acknowledged by its peer.
-However, this does not guarantee that no further packets will need to be
-received or sent at that encryption level because a peer might not have received
-all the acknowledgments necessary to reach the same state.
+An endpoint cannot discard keys for a given encryption level unless it has
+received all CRYPTO frames for that encryption level, acknowledged the packets
+containing those frames, and received acknowledgements for packets containing
+all CRYPTO frames sent for that encryption level. However, this does not
```
Comment (marten-seemann) [inline draft-ietf-quic-tls.md:842]: Is that so? I could receive HANDSHAKE_DONE without having received all ACKs for Handshake packets, right?

Comment (martinthomson) [reply draft-ietf-quic-tls.md:842]: In that case, yes.  And I guess we also have other shortcuts (you don't need Initial ACKs to discard Initial keys as a client or server).

**Review by janaiyengar — APPROVED** [review]
With @martinthomson's fixes, this is good.

**Pull request merged**